includes/blocks/class-sensei-quiz-blocks.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * File containing the class Sensei_Quiz_Blocks.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Class Sensei_Quiz_Blocks
+ */
+class Sensei_Quiz_Blocks extends Sensei_Blocks_Initializer {
+
+	/**
+	 * Course progress block.
+	 *
+	 * @var Sensei_Block_Quiz
+	 */
+	public $quiz;
```

Comment:
As discussed earlier, I don't think that we should expose these properties as public.

Reply:
These are public in the spirit of keeping Sensei extensible — this is the only way to access these instances, eg. if someone wants to unregister some hooks or needs to reference them in any way. This continues the pattern used everywhere else in Sensei of exposing the singleton instances through the `Sensei_Main` global.

Reply:
> These are public in the spirit of keeping Sensei extensible — this is the only way to access these instances, eg. if someone wants to unregister some hooks or needs to reference them in any way.

Extensibility shouldn't be achieved in the cost of encapsulation. If we want to allow extensibility in some object we should do it by using hooks (which is the WP way) or abstractions.

> This continues the pattern used everywhere else in Sensei of exposing the singleton instances through the Sensei_Main global.

I think that this is [widely considered](https://en.wikipedia.org/wiki/Object_orgy) as an anti-pattern. We shouldn't repeat existing problems of the codebase.